src/test/java/org/jenkinsci/plugins/workflow/properties/PipelineTriggersJobPropertyTest.java

```diff
@@ -0,0 +1,177 @@
+import org.jenkinsci.plugins.workflow.job.properties.PipelineTriggersJobProperty;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+import org.jvnet.hudson.test.recipes.LocalData;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.ObjectStreamException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class PipelineTriggersJobPropertyTest {
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void loadCallsStartFalse() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        p = (WorkflowJob) Items.load(p.getParent(), p.getRootDir());
+        t = (MockTrigger)p.getTriggers().get(t.getDescriptor());
+        assertNotNull(t);
+        assertEquals("[false]", t.calls.toString());
+        assertTrue(t.isStarted);
+    }
+
+    @Test
+    public void submitCallsStartTrue() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        p = (WorkflowJob) r.configRoundtrip((Item)p);
+        t = (MockTrigger)p.getTriggers().get(t.getDescriptor());
+        assertNotNull(t);
+        assertEquals("[true]", t.calls.toString());
+        assertTrue(t.isStarted);
+    }
+
+    @Test
+    public void previousTriggerStopped() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        MockTrigger t = new MockTrigger();
+        p.addTrigger(t);
+        p.save();
+        assertTrue(t.isStarted);
+        MockTrigger t2 = new MockTrigger();
+        p.addTrigger(t2);
+        p.save();
+
+        assertFalse(t.isStarted);
+        assertTrue(t2.isStarted);
+    }
+
+    @LocalData
+    @Test
+    public void triggerMigration() throws Exception {
+        WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+        assertNotNull(p);
+
+        PipelineTriggersJobProperty triggerProp = p.getProperty(PipelineTriggersJobProperty.class);
+        assertNotNull(triggerProp);
+        assertEquals(1, triggerProp.getTriggers().size());
+        assertEquals(1, p.getTriggers().size());
+
+        Trigger fromProp = triggerProp.getTriggers().get(0);
+        assertEquals(TimerTrigger.class, fromProp.getClass());
+
+        Trigger fromJob = p.getTriggers().get(fromProp.getDescriptor());
+        assertEquals(fromProp, fromJob);
```

Comment:
Will check - it's definitely only getting called once in `loadCallsStartFalse` and `submitCallsStartTrue` or they'd be failing on the `t.calls.toString()` assert.